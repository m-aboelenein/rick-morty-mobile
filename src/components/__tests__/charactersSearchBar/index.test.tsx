import {fireEvent, render} from '@testing-library/react-native';
import * as React from 'react';
import renderer from 'react-test-renderer';
import CharacterSearchBar from '../../charactersSearchBar';

const mockSetSearchValue = jest.fn();

it('renders a character search bar', () => {
  const tree = renderer.create(
    <CharacterSearchBar
      searchValue="rick"
      setSearchValue={mockSetSearchValue}
    />,
  );
  expect(tree).toMatchSnapshot();
});

it('searches for characters with the right values', () => {
  const {getByTestId} = render(
    <CharacterSearchBar
      searchValue="rick"
      setSearchValue={mockSetSearchValue}
    />,
  );

  fireEvent.changeText(getByTestId('search-bar'), 'rick');
  expect(mockSetSearchValue).toHaveBeenCalled();
  expect(mockSetSearchValue).toBeCalledWith('rick');
});
