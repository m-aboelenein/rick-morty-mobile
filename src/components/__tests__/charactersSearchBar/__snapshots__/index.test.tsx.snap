// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a character search bar 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    onChangeText={[MockFunction]}
    placeholder="Enter Character Name"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 8,
        "color": "black",
        "height": 40,
        "paddingHorizontal": 8,
      }
    }
    testID="search-bar"
    underlineColorAndroid="transparent"
    value="rick"
  />
</View>
`;
